conversant/demo/ui.py

```diff
@@ -16,40 +16,71 @@
 from conversant.demo import utils
 
 
+def render_bot_partial_reply(utterance, idx):
+    """Renders a partial reply message from the bot.
+
+    Args:
+        turn (str): The utterance to be rendered.
+        idx (int): The index of the turn.
+    """
+    st_message(
+        value=utterance.rstrip(),
+        animate_from=""
+        if "prev_partial_chunk" not in st.session_state
+        else st.session_state.prev_partial_chunk.lstrip(),
+        use_typewriter=True,
+        key=f"{idx}_bot",
+        avatar_style=st.session_state.bot_avatar,
+        generation_complete=(not st.session_state.partial_reply_in_progress),
+    )
+
+
 def draw_chat_history() -> None:
     """Renders the chat history in Streamlit.
 
     The messages are rendered using streamlit-chat, a custom Streamlit component
     for a chatbot UI.
     Reference: https://github.com/AI-Yash/st-chat
     """
-    # The chat history is iterated in reverse order to ensure that recent messages
-    # displayed are anchored at the bottom of the Streamlit demo.
-    for i, turn in enumerate(reversed(st.session_state.bot.chat_history)):
+    for i, turn in enumerate(st.session_state.bot.chat_history):
 
-        # If we are at the first conversation turn, we remove the
-        # injected user utterance of "Hello" from displaying.
-        if i == len(st.session_state.bot.chat_history) - 1:
+        # If there is only one turn, then we should only show the
+        # bot utterance (but using the typewriter and partial reply effect),
+        # skipping over the first injected user utterance.
+        if len(st.session_state.bot.chat_history) == 1:
             if "bot" in turn:
+                render_bot_partial_reply(turn["bot"], i)
+
+        # If we are at the last conversation turn, the bot utterance
+        # will be rendered as a partial reply with the typewriter effect.
+        elif i == len(st.session_state.bot.chat_history) - 1:
+            if "user" in turn:
                 st_message(
-                    turn["bot"],
-                    key=f"{i}_bot",
-                    avatar_style=st.session_state.bot_avatar,
+                    value=turn["user"],
+                    animate_from="",
```

Comment:
Why is this call to `st_message` different from the one in lines 72 to 77? In that this one contains `animate_from=""` but the other one does not.

Reply:
The default arg is `animate_from=""` so this isn't necessary. Removed in [f066a01](https://github.com/cohere-ai/sandbox-conversant-lib/pull/99/commits/f066a01b04fd61ae86790c5bdba7eada7eccb397)